asm/Game/zNPCHazard.s

```diff
@@ -755,8 +755,11 @@ lbl_801881F8:
 /* 80188204 00185004  38 21 00 10 */	addi r1, r1, 0x10
 /* 80188208 00185008  4E 80 00 20 */	blr 
 
-.global HAZ_Iterate__FPFR9NPCHazardPv_bPvi
-HAZ_Iterate__FPFR9NPCHazardPv_bPvi:
+/* changed from ... */
+/* HAZ_Iterate__FPFR9NPCHazardPv_bPvi */
+/* ... so linker can find it */
+.global HAZ_Iterate__FPFR9NPCHazardPv_UcPvi
+HAZ_Iterate__FPFR9NPCHazardPv_UcPvi:
```

Comment:
The linker couldn't find HAZ_Iterate because its declaration in zNPCHazard.h was incorrect (DWARF treats bool as uint8 for whatever reason).

`void HAZ_Iterate(uint8 (*fp)(NPCHazard&, void*), void* context, int32 flag_filter);` should be changed to:
`void HAZ_Iterate(bool (*fp)(NPCHazard&, void*), void* context, int32 flag_filter);`

Then change the HAZ_Iterate symbol back here and in zNPCHazard.cpp, and the linker should be able to find it.